R/declare_test.R

```diff
@@ -78,13 +80,64 @@
 #' )
 #' }
 #' 
-declare_test <- declare_estimator
+#' @seealso See \code{\link{declare_estimator}} for documentation of the \code{model_handler} function.
+#' 
+declare_test <- 
+  make_declarations(
+    tidy_test(model_handler),
+    step_type = "estimator",
+    causal_type = "estimator",
+    default_label = "test"
+  )
+
 declare_tests <- declare_test
 
 #' @details
-#' \code{tidy_test} takes an untidy test function, and returns a tidy handler which accepts standard labeling options.
+#' \code{tidy_test} takes a data-in-data out function to \code{fn}, and returns a data-in-data-out function that first runs the provided test function \code{fn} and then appends a label for the test.
 #'
-#' @param fn A function that takes a data.frame as an argument and returns a data.frame with the test statistic (and other inferential statistics) and a label.
+#' @param fn A function that takes a data.frame as an argument and returns a data.frame with test statistics as columns.
 #' @rdname declare_test
 #' @export
-tidy_test <- tidy_estimator
+tidy_test <- function(fn) {
+  if (!("data" %in% names(formals(fn)))) {
+    stop("Must provide a `test_function` function with a data argument.")
```

Comment:
If the only difference is the error message, you can parameterize the string instead of duplicate the code.

Reply:
They're no longer the same so not doing this for now but will think about how to simplify for future